src/tools/cephfs/cephfs-shell

```diff
@@ -773,6 +773,129 @@ class CephFSShell(Cmd):
 
         return 0
 
+    def complete_ln(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    ln_parser = argparse.ArgumentParser(
+        description='Create a hard link or a symbolic '
+                    'link to an existing file or directory.')
+    ln_parser.add_argument('source_path', type=str, action=path_to_bytes,
+                           help='Path of the source file')
+    ln_parser.add_argument('destination_path', type=str, action=path_to_bytes,
+                           help='Destination path', nargs='?')
+    ln_parser.add_argument('-s', '--symbolic', action='store_true',
+                           help='Create symbolic link')
+    ln_parser.add_argument('-v', '--verbose', action='store_true',
+                           help='Print name of each linked file')
+    ln_parser.add_argument('-f', '--force', action='store_true',
+                           help='Force create link/symbolic link')
+
+    @with_argparser(ln_parser)
+    def do_ln(self, args):
+        # check if source file/dir exists.
+        if not is_file_exists(args.source_path) \
+                and not is_dir_exists(args.source_path):
+            set_exit_code_msg(errno.ENOENT,
+                              msg=f"ln: failed to access "
+                                  f"'{args.source_path.decode('utf-8')}"
+                                  f"': No such file or directory")
+            return
+        source_path = args.source_path
+
+        destination_path = ''
+        # trying to create link at cwd
+        if args.destination_path is None:
+            cwd = cephfs.getcwd().decode('utf-8')
+            if cwd[len(cwd) - 1] != '/':
+                cwd = cwd + '/'
+            cwd = cwd.encode('utf-8')
+            print('cwd: ', cwd)
+            paths = args.source_path.decode('utf-8').split('/')
+
+            if (is_file_exists(args.source_path) or is_dir_exists(args.source_path)) \
+                    and len(paths[len(paths) - 1]) != 0:
+                source_file_or_dir = paths[len(paths) - 1]
+            else:
+                source_file_or_dir = paths[len(paths) - 2]
+            destination_path = cwd + source_file_or_dir.encode('utf-8')
+
+            if (is_file_exists(destination_path) or is_dir_exists(destination_path)) \
+                    and not args.force:
+                set_exit_code_msg(errno.ENOENT,
+                                  msg=f"ln: failed to create link "
+                                      f"{destination_path.decode('utf-8')}: "
+                                      f"exists")
+                return
+        # time to check for destination path provided
```

Comment:
nit: redundant comment?

Reply:
Yes, Ill also need to clean a few other comments too